tools/releaseBuild/azureDevOps/templates/updateBuildTable.yml

```diff
@@ -0,0 +1,48 @@
+jobs:
+- job: updateBuildTable
+  displayName: Update Build Information Tbale
+  dependsOn:
+    ${{ parameters.parentJobs }}
+  
+  # the package EZ pools don't have the latest module needed to run this job
+  pool:
+    name: Hosted Mac Internal
+
+  steps:
+  - template: SetVersionVariables.yml
+    parameters:
+      ReleaseTagVar: $(ReleaseTagVar)
+
+  - powershell: |
+      install-module azstoragetable.travisez13 -Scope CurrentUser -Force -SkipPublisherCheck
+    displayName: 'Install Azure Table Module'
+
+  - powershell: |
+      $az_key = ConvertTo-SecureString -Force -String '$(az_key)' -AsPlainText
+      $az_key_cred = [PSCredential]::new('$(az_appid)',$az_key)
+
+      Write-Verbose "Connecting to Azure ..." -verbose
+      Connect-AzAccount -ServicePrincipal -Credential $az_key_cred -TenantId '$(az_tenantid)'   -force -scope process
+
+      $table = Get-AzStorageTableTable -resourceGroup pscoredata -storageAccountName pscorebuilddata -tableName buildData
+      
+      Write-Verbose "Removing existing row, if it exists ..." -verbose
+      $row = Remove-AzStorageTableRow -table $table -partitionkey '$(System.DefinitionId)' -rowKey '$(ReleaseTagVar)'
+
+      Write-Verbose "Adding new row ..." -verbose
+      $result = Add-AzStorageTableRow -table $table -partitionkey '$(System.DefinitionId)' -rowKey '$(ReleaseTagVar)' -property @{
+              buildid = '$(Build.BuildId)'
+      
+            }
+      While(!$result.IsCompleted)
+      {
+        Write-Verbose -message "waiting for result" -verbose
+        Start-Sleep -Seconds 1
+      }
+
+      if(!$result.IsCompletedSuccessfully)
```

Comment:
Do we need a timeout?

Reply:
The whole job has a timeout, which should catch before compliance finishes.

I'll lowe it to 15 minutes.